force-app/main/default/classes/CBCreateSubscriptionTest.cls

```diff
@@ -1,397 +1,473 @@
-                 {
-                     string pid = pr.Id;
-             		 selectedIds.add(pid);
-                 }
-             }
-             if((plantype.equalsIgnoreCase('Tiered') || plantype.equalsIgnoreCase('StairStep')||plantype.equalsIgnoreCase('Volume')))
+   		chargebeeapps__CB_Sites__c defaultSite=TestDataUtility.getSite();
+    	Account[] accountslist=TestDataUtility.getAccounts();
+    	Account oppAccount=accountslist[0];
+    	List<Opportunity> testOpportunity=TestDataUtility.getOpportunities(oppAccount.Id);   
+	    List<Contact> oppContacts=TestDataUtility.getContactFromAccount(oppAccount.Id);
+	    Contact oppContact=oppContacts[0];
+        OpportunityLineItem opplineitem = TestDataUtility.getOppLineItem(testOpportunity[0].Id);
+        TestDataUtility.associateContactToSite(defaultSite.Id,oppContact.Id);
+	    PageReference pr = new PageReference('/apex/Create_Subscription');
+		pr.getParameters().put('opp_id',testOpportunity[0].Id);
+		pr.getParameters().put('con_id',oppContact.Id);
+		//pr.getParameters().put('domain',defaultSite.Id);
+		Test.setCurrentPageReference(pr);
+	    CBCreateSubscription cbs=new CBCreateSubscription();
+        // SiteDetails autofill Validation
+        List<selectOption> siteList=cbs.getSiteDetails();
+        System.assert(siteList.size() == 1);
+        System.assert(cbs.isSiteConnected);
+        System.assert(siteList[0].getLabel() == defaultSite.Name);        
+        // Opportunity and Contact auot fill Validation        
+	    PageReference noerror=cbs.checkPermissions();        
+	    System.debug('Error page ' + noerror);                
+        System.debug(' ContactName Expected ' + cbs.con_id + ' Actual ' + oppContact.Id );
+        System.debug(' OpportunityName Expected ' + cbs.opp_id + ' Actual ' + testOpportunity[0].Id );
+        System.debug('IsSiteSelected ' + cbs.isSiteConnected + 'SelectedSiteName ' + cbs.selectedSites);        
+	    System.assert(noerror == null); // When permission pass the test, return value is null
+        System.assert(cbs.opp_id != null);
+        System.assertEquals(cbs.opp_id, testOpportunity[0].Id);        
+	    System.assertEquals(cbs.con_id, oppContact.Id);
+        System.assert(cbs.oppOptions == null); 
+        // Plans and Addon Autofill validation
+       	System.debug('DefaultSiteId input to fetch PBEntries ' + defaultSite.Id + ' Name ' + defaultSite.Name);
+        List<PriceBookEntry> pbEntriesActual=TestDataUtility.getPBEntryBySite(defaultSite.Id,'PLAN');                        
+        System.debug(' ExpectedPlanConented Content ' + cbs.options3);
+        System.debug(' pbEntriesActual->Size ' + pbEntriesActual.size() + ' plansExpected->SIze ' + cbs.options3.size() );
+        System.assert(comparePlansSelectList(cbs.options3,pbEntriesActual)); // Plan selectlist from Controller matches the testdata       
+    }
```

Comment:
validate auto fill of plans and addons

Reply:
Autofill validation done for Plans and Addons